freqtrade/optimize/__init__.py

```diff
@@ -112,12 +139,31 @@ def download_pairs(datadir, pairs: List[str], ticker_interval: str) -> bool:
     return True
 
 
+def get_start_ts_from_timerange(timerange: Tuple[Tuple, int, int], interval: str) -> int:
+    if not timerange:
+        return None
+
+    if timerange[0][0] == 'date':
+        return timerange[1] * 1000
+
+    if timerange[0][1] == 'line':
+        num_minutes = timerange[2] * Constants.TICKER_INTERVAL_MINUTES[interval]
+        return arrow.utcnow().shift(minutes=num_minutes).timestamp * 1000
+
+    return None
+
+
 # FIX: 20180110, suggest rename interval to tick_interval
```

Comment:
not directly related to this PR - but if we touch this function - should we also consider applying this comment?

Reply:
will do